### PR TITLE
Fix json_parse for large number

### PR DESCRIPTION
**Pull request by Yuhta**
Summary:
When the number is well-formated but outside the range of int64, we
still allow such cases in `json_parse` so that we can cast them to floating
point numbers later.  This is achieved by replacing the DOM parser with a tree
walk visitor with on-demand parser.  Also relax the corresponding number parsing
in JSON cast.

Fix https://github.com/facebookincubator/velox/issues/8066

Differential Revision: D52888389

### CODE REVIEW

File: CMake/resolve_dependency_modules/simdjson.cmake [hunk 1]
```diff
@@ -13,9 +13,9 @@
 # limitations under the License.
 include_guard(GLOBAL)
 
-set(VELOX_SIMDJSON_VERSION 3.1.5)
```
Comment (mbasmanova) [inline CMake/resolve_dependency_modules/simdjson.cmake:16]: Is this an unrelated upgrade? If yes, please, move it into a separate PR. Otherwise, include it in PR description.

File: velox/functions/prestosql/JsonFunctions.cpp [hunk 2]
```diff
@@ -71,23 +71,21 @@ class JsonParseFunction : public exec::VectorFunction {
...
-    // validation of JSON syntax that doesn't allocate memory or copy data.
     assert(args.size() > 0);
     const auto& arg = args[0];
     if (arg->isConstantEncoding()) {
       auto value = arg->as<ConstantVector<StringView>>()->valueAt(0);
-      // Ask the parser to copy the input in case `value' is inlined.
-      auto parsed = parser_.parse(value.data(), value.size(), true);
-      if (parsed.error() != simdjson::SUCCESS) {
-        context.setErrors(rows, errors_[parsed.error()]);
+      paddedInput_.resize(value.size() + simdjson::SIMDJSON_PADDING);
```
Comment (mbasmanova) [inline velox/functions/prestosql/JsonFunctions.cpp:85]: Unrelated change?

**Review by mbasmanova — APPROVED** [review]
Thanks.

File: velox/functions/prestosql/JsonFunctions.cpp (hunk 2 above)
Comment (Yuhta) [reply velox/functions/prestosql/JsonFunctions.cpp:85]: It's needed, the DOM parser was doing this in their code, and now because we swapped the DOM parser with ondemand one, we need to do the padding ourselves.
